packages/grafana-schema/src/schema/dashboard/dashboard.gen.ts

```diff
@@ -0,0 +1,222 @@
+// This file is autogenerated. DO NOT EDIT.
+//
+// To regenerate, run "make gen-cue" from repository root.
+//
+// Derived from the Thema lineage at pkg/coremodel/dashboard
+
+
+// This model is a WIP and not yet canonical. Consequently, its members are
+// not exported to exclude it from grafana-schema's public API surface.
+
+interface AnnotationQuery {
+  builtIn: number;
+  datasource: {};
+  enable: boolean;
+  hide?: boolean;
+  iconColor?: string;
+  name?: string;
+  rawQuery?: string;
+  showIn: number;
+  target?: {};
+  type: string;
+}
+
+const defaultAnnotationQuery: Partial<AnnotationQuery> = {
+  builtIn: 0,
+  enable: true,
+  hide: false,
+  showIn: 0,
+  type: 'dashboard',
+};
+
+interface VariableModel {
+  label?: string;
+  name: string;
+  type: VariableType;
+}
+
+interface DashboardLink {
+  asDropdown: boolean;
+  icon?: string;
+  includeVars: boolean;
+  keepTime: boolean;
+  tags: string[];
+  targetBlank: boolean;
+  title: string;
+  tooltip?: string;
+  type: DashboardLinkType;
+  url?: string;
+}
+
+const defaultDashboardLink: Partial<DashboardLink> = {
+  asDropdown: false,
+  includeVars: false,
+  keepTime: false,
+  tags: [],
+  targetBlank: false,
+};
+
+type DashboardLinkType = 'link' | 'dashboards';
+
+type VariableType = 'query' | 'adhoc' | 'constant' | 'datasource' | 'interval' | 'textbox' | 'custom' | 'system';
+
+enum FieldColorModeId {
+  ContinuousGrYlRd = 'continuous-GrYlRd',
+  Fixed = 'fixed',
+  PaletteClassic = 'palette-classic',
+  PaletteSaturated = 'palette-saturated',
+  Thresholds = 'thresholds',
+}
+
+type FieldColorSeriesByMode = 'min' | 'max' | 'last';
+
+interface FieldColor {
+  fixedColor?: string;
+  mode: FieldColorModeId | string;
+  seriesBy?: FieldColorSeriesByMode;
+}
+
+interface Threshold {
+  color: string;
+  state?: string;
+  value?: number;
+}
+
+enum ThresholdsMode {
+  Absolute = 'absolute',
+  Percentage = 'percentage',
+}
+
+interface ThresholdsConfig {
+  mode: ThresholdsMode;
+  steps: Threshold[];
+}
+
+const defaultThresholdsConfig: Partial<ThresholdsConfig> = {
+  steps: [],
+};
+
+interface Transformation {
+  id: string;
+  options: {};
+}
+
+enum DashboardCursorSync {
+  Crosshair = 1,
+  Off = 0,
+  Tooltip = 2,
+}
+
+const defaultDashboardCursorSync: DashboardCursorSync = DashboardCursorSync.Off;
+
+interface Panel {
+  datasource?: {};
+  description?: string;
+  fieldConfig: {
+    defaults: {};
+    overrides: {
```

Comment:
I am still very wary of merging the frontend interfaces that are impossible to use/test (as typescript)

What is the value of merging?  is it so that `git diff` shows what changes?

How does, or how can this relate to the `mudball.cue` > `graph.gen.ts` functions?  At least there we manually edit the generated files to what needs to be created so we can use them

Reply:
> I am still very wary of merging the frontend interfaces

What's the bad outcome to be avoided here?

> What is the value of merging? is it so that git diff shows what changes?

That's the one i said before, and yes, it's still my primary focus. Working on fixing cuetsy impressed on me the importance of that. Codegen needs to be obvious and fade into the background to work well.

There's also two other reasons.

My feeling is that, for both mobbing and async discussion, it works better to have a concrete thing out there, even if it's wrong. There's a lot of degrees of freedom here, and removing one helps counter analysis paralysis. Making issues/TODOs/follow-ups is easy for code that exists but is wrong, but near-impossible for code that _would_ exist if you ran the generator.

Last is comms. Schematization/coremodels has enough actual working code now to write product + design docs that i think folks will find understandable. But it's easier to make those explanations with links to the code that's derived from schema. Same deal for a blog post i'd like to write related to as-code tooling, which i know users have an eye out for with 9.

> How does, or how can this relate to the mudball.cue > graph.gen.ts functions? At least there we manually edit the generated files to what needs to be created so we can use them

The fix to get them generating is trivial now. I left those out here because they already _are_ in use, which means that they need to be addressed by someone with actual expertise, through some pairing/mobbing.

Though...looking at `graph.gen.ts`, i might be able to just bring the CUE in line with the generated TS, and can call it done. Will give some minutes tonight